packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m

```diff
@@ -71,72 +73,82 @@ - (instancetype)init
   if (self = [super init]) {
     _extraData = [NSMutableArray new];
     _extraDataTitle = [NSMutableArray new];
-    self.view.backgroundColor = [UIColor colorWithRed:0.8 green:0 blue:0 alpha:1];
+  }
+  return self;
+}
 
-    _tableView = [UITableView new];
-    _tableView.delegate = self;
-    _tableView.dataSource = self;
-    _tableView.backgroundColor = [UIColor clearColor];
-    _tableView.estimatedRowHeight = 200;
-    _tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
-    _tableView.rowHeight = UITableViewAutomaticDimension;
-    _tableView.allowsSelection = NO;
+- (UIButton *)redBoxButtonWithTitle:(NSString *)title accessibilityIdentifier:(NSString *)accessibilityIdentifier action:(SEL)action {
+  UIButton *button = [UIButton buttonWithType: UIButtonTypeCustom];
+  button.accessibilityIdentifier = accessibilityIdentifier;
+  
+  button.backgroundColor = [UIColor colorWithRed:0.1 green:0.1 blue:0.1 alpha:1];
+  
+  button.titleLabel.font = [UIFont systemFontOfSize:13];
+  
+  [button setTitle:title forState:UIControlStateNormal];
+  [button setTitleColor:[UIColor colorWithWhite:1 alpha:1] forState:UIControlStateNormal];
+  [button setTitleColor:[UIColor colorWithWhite:1 alpha:0.5] forState:UIControlStateHighlighted];
+  
+  [button addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
+  return button;
+}
 
+-(void)viewDidLoad {
+  CGFloat bottomSafeAreaInset = RCTSharedApplication().delegate.window.safeAreaInsets.bottom;
+  
+  CGFloat buttonHeight = 60;
+  CGFloat buttonWidth = self.view.bounds.size.width / 2;
+  
+  CGRect frame = self.view.bounds;
+  frame.size.height -= buttonHeight + bottomSafeAreaInset;
+  
+  _tableView = [[UITableView alloc] initWithFrame:frame style:UITableViewStylePlain];
+  _tableView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
```

Comment:
How about making this table view also use auto layout?

Reply:
Agree!